pkg/registry/provider_remote.go

```diff
@@ -58,62 +58,106 @@ func (p *RemoteRegistryProvider) GetRegistry() (*Registry, error) {
 	for name, server := range registry.Servers {
 		server.Name = name
 	}
+	// Set name field on each remote server based on map key
```

Comment:
It looks like `provider_remote` and `provider_local` are mostly the same. It would be good if we could split out the common methods from the interface.